### PR TITLE
[cdn]function Cdn migration to afd added

### PR DESCRIPTION
**Pull request by Ptnan7**
## Mandatory Checklist

- Please choose the target release of Azure PowerShell. (⚠️**Target release** is a different concept from **API readiness**. Please click below links for details.)
  - [ ] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] No need for a release

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by t-bzhan** [comment]
@Ptnan7 , can you correcte the grammar in the PR title and clearly expressed the intention to add cmdlets to support classical CDN migration.

File: src/Cdn/Cdn.Autorest/custom/Move-AzCdnProfileToAFD.ps1 [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+
+MIGRATIONPARAMETER <ICdnMigrationToAfdParameters>: Request body for Migrate operation.
+  [MigrationEndpointMapping <IMigrationEndpointMapping[]>]: A name map between classic CDN endpoints and AFD Premium/Standard endpoints.
+    [MigratedFrom <String>]: The name of the old endpoint.
+    [MigratedTo <String>]: The name for the new endpoint.
+  [SkuName <SkuName?>]: Name of the pricing tier.
+.Link
+https://learn.microsoft.com/powershell/module/az.cdn/move-azcdnprofiletoafd
+#>
+function Move-AzCdnProfileToAFD {
```
Comment (t-bzhan) [inline src/Cdn/Cdn.Autorest/custom/Move-AzCdnProfileToAFD.ps1:68]: Could you please sync with the PS team to determine whether this name is suitable for our migration purpose? I remember we used to have a process to review command names/parameters in advance, but I’m not sure if it’s still needed.

File: src/Cdn/Cdn.Autorest/custom/Invoke-AzCdnCommitProfileToAFDMigration.ps1 [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        [Parameter(DontShow)]
+        [Microsoft.Azure.PowerShell.Cmdlets.Cdn.Category('Runtime')]
+        [System.Management.Automation.SwitchParameter]
+        # Use the default credentials for the proxy
+        ${ProxyUseDefaultCredentials}
+    )
+
+    process {
+        Write-Host("Start to migrate.")
+        Write-Host("This process will disable your Front Door (classic) profile and move all your traffic and configurations to the new Front Door profile.")
```
Comment (t-bzhan) [inline src/Cdn/Cdn.Autorest/custom/Invoke-AzCdnCommitProfileToAFDMigration.ps1:124]: Copy typo, this is not for Classical Front Door.

File: src/Cdn/Cdn.Autorest/custom/Move-AzCdnProfileToAFD.ps1 [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+
+        Write-Host("Now you can commit the migration to finalize the migration process.")
+    }
+    }
+    
+function ValidateIdentityType {
+    if (${IdentityType}) {
+        $identityTypeArray =  ${IdentityType}.ToString().split(",")
+        if (($identityTypeArray.Count -gt 2)) {
+            throw "The IdentityType is invalid. The supported types are 'SystemAssigned,UserAssigned' when the front door has Customer Certificates during migration."
```
Comment (t-bzhan) [inline src/Cdn/Cdn.Autorest/custom/Move-AzCdnProfileToAFD.ps1:243]: Copy typo

File: src/Cdn/Cdn.Autorest/docs/Invoke-AzCdnCommitProfileToAFDMigration.md [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+## SYNTAX
+
+```
+Invoke-AzCdnCommitProfileToAFDMigration -ProfileName <String> -ResourceGroupName <String>
+ [-SubscriptionId <String>] [-DefaultProfile <PSObject>] [-AsJob] [-NoWait] [-PassThru] [-Confirm] [-WhatIf]
+ [<CommonParameters>]
+```
+
+## DESCRIPTION
+Commit the migrated Azure Front Door(Standard/Premium) profile..
```
Comment (t-bzhan) [inline src/Cdn/Cdn.Autorest/docs/Invoke-AzCdnCommitProfileToAFDMigration.md:22]: Could you please provide more details regarding what this step means? Additionally, what are the consequences if a customer conducts this step?

File: src/Cdn/Cdn.Autorest/examples/Invoke-AzCdnCommitProfileToAFDMigration.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
+### Example 1: Commit the Migration
+```powershell
+Invoke-AzCdnCommitProfileToAFDMigration -ProfileName name-migrated -ResourceGroupName rgName
+```
+
+Commit the Migration
+
+### Example 2: Commit the migrated Azure Front Door(Standard/Premium) profile, when the subscription of the profile is different from the local subscrition
```
Comment (t-bzhan) [inline src/Cdn/Cdn.Autorest/examples/Invoke-AzCdnCommitProfileToAFDMigration.md:8]: That is somewhat confusing. Are we allowed to migrate to another subscription?

File: src/Cdn/Cdn.Autorest/test/Get-AzCdnKeyGroup.Tests.ps1 [hunk 6]
```diff
@@ -0,0 +1,29 @@
+if(($null -eq $TestName) -or ($TestName -contains 'Get-AzCdnKeyGroup'))
+{
```
Comment (t-bzhan) [inline src/Cdn/Cdn.Autorest/test/Get-AzCdnKeyGroup.Tests.ps1:2]: KeyGroup support is not ready yet, suggest to remove it.

File: src/Cdn/Cdn.Autorest/Az.Cdn.psd1
Comment (t-bzhan) [inline src/Cdn/Cdn.Autorest/Az.Cdn.psd1:1]: Please update the changelog

**Review by t-bzhan — CHANGES_REQUESTED** [review]

**Comment by VeryEarly** [comment]
API has been changed to a preview version, is it intended or this PR is targeting a preview release?

**Comment by Ptnan7** [comment]
> API has been changed to a preview version, is it intended or this PR is targeting a preview release?

Yes, this PR is a preview release

**Comment by VeryEarly** [comment]
> > API has been changed to a preview version, is it intended or this PR is targeting a preview release?
> 
> Yes, this PR is a preview release

do you understand what preview in Azure-PowersShell means? please read the link in the checklist of this PR

**Comment by Ptnan7** [comment]
> > > API has been changed to a preview version, is it intended or this PR is targeting a preview release?
> > 
> > 
> > Yes, this PR is a preview release
> 
> do you understand what preview in Azure-PowersShell means? please read the link in the checklist of this PR

Out team want to release a preview api, could you help us to create a preview branch for us?

**Pull request merged**